### DISCUSSION TITLE
Error installing coursework

### DISCUSSION THREAD

**Discussion by joyalrj22** [wenceorg/comp4187#30] [Q&A] [answered]
I tried to pip install the coursework repo using pip install -e mg-202021-joyalrj22 and it fails with thee following output:
(Note: petsc is already installed into a virtualenv and this virtualnv was activated)

`Obtaining file:///mnt/c/Users/Administrator/Documents/parallell2/mg-202021-joyalrj22
Requirement already satisfied: Cython in ./psc/lib/python3.6/site-packages (from mgsolver==0.1)
Requirement already satisfied: flake8 in ./psc/lib/python3.6/site-packages (from mgsolver==0.1)
Collecting numba (from mgsolver==0.1)
  Using cached https://files.pythonhosted.org/packages/e3/7d/3d61160836e49f40913741c464f119551c15ed371c1d91ea50308495b93b/numba-0.53.1.tar.gz
Requirement already satisfied: numpy in ./psc/lib/python3.6/site-packages (from mgsolver==0.1)
Collecting petsc4py (from mgsolver==0.1)
  Using cached https://files.pythonhosted.org/packages/c1/73/604fe37a9305586ad13807ffc81f2a0535f5c0a2cd2c615f248f769374c0/petsc4py-3.14.1.tar.gz
Requirement already satisfied: pytest in ./psc/lib/python3.6/site-packages (from mgsolver==0.1)
Requirement already satisfied: pycodestyle<2.8.0,>=2.7.0 in ./psc/lib/python3.6/site-packages (from flake8->mgsolver==0.1)
Requirement already satisfied: importlib-metadata; python_version < "3.8" in ./psc/lib/python3.6/site-packages (from flake8->mgsolver==0.1)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in ./psc/lib/python3.6/site-packages (from flake8->mgsolver==0.1)
Requirement already satisfied: pyflakes<2.4.0,>=2.3.0 in ./psc/lib/python3.6/site-packages (from flake8->mgsolver==0.1)
Collecting llvmlite<0.37,>=0.36.0rc1 (from numba->mgsolver==0.1)
  Using cached https://files.pythonhosted.org/packages/19/66/6b2c49c7c68da48d17059882fdb9ad9ac9e5ac3f22b00874d7996e3c44a8/llvmlite-0.36.0.tar.gz
Requirement already satisfied: setuptools in ./psc/lib/python3.6/site-packages (from numba->mgsolver==0.1)
Requirement already satisfied: petsc<3.15,>=3.14 in ./psc/lib/python3.6/site-packages (from petsc4py->mgsolver==0.1)
Requirement already satisfied: py>=1.8.2 in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: toml in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: attrs>=19.2.0 in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: pluggy<1.0.0a1,>=0.12 in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: packaging in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: iniconfig in ./psc/lib/python3.6/site-packages (from pytest->mgsolver==0.1)
Requirement already satisfied: typing-extensions>=3.6.4; python_version < "3.8" in ./psc/lib/python3.6/site-packages (from importlib-metadata; python_version < "3.8"->flake8->mgsolver==0.1)
Requirement already satisfied: zipp>=0.5 in ./psc/lib/python3.6/site-packages (from importlib-metadata; python_version < "3.8"->flake8->mgsolver==0.1)
Requirement already satisfied: pyparsing>=2.0.2 in ./psc/lib/python3.6/site-packages (from packaging->pytest->mgsolver==0.1)
Building wheels for collected packages: numba, petsc4py, llvmlite
  Running setup.py bdist_wheel for numba ... error
  Complete output from command /mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qo8wd315/numba/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpe426nrg1pip-wheel- --python-tag cp36:
  TBB not found
  Using OpenMP from: True
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for numba
  Running setup.py clean for numba
  Running setup.py bdist_wheel for petsc4py ... error
  Complete output from command /mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qo8wd315/petsc4py/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp0pstg7wspip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for petsc4py
  Running setup.py clean for petsc4py
  Running setup.py bdist_wheel for llvmlite ... error
  Complete output from command /mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qo8wd315/llvmlite/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpmwwpevo2pip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help

  error: invalid command 'bdist_wheel'

  ----------------------------------------
  Failed building wheel for llvmlite
  Running setup.py clean for llvmlite
Failed to build numba petsc4py llvmlite
Installing collected packages: llvmlite, numba, petsc4py, mgsolver
  Running setup.py install for llvmlite ... error
    Complete output from command /mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qo8wd315/llvmlite/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-ot_v27hb-record/install-record.txt --single-version-externally-managed --compile --install-headers /mnt/c/Users/Administrator/Documents/parallell2/psc/include/site/python3.6/llvmlite:
    running install
    running build
    got version from file /tmp/pip-build-qo8wd315/llvmlite/llvmlite/_version.py {'version': '0.36.0', 'full': 'e6bb8d137d922bec8beeb01a237254778759becd'}
    running build_ext
    /mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 /tmp/pip-build-qo8wd315/llvmlite/ffi/build.py
    LLVM version... Traceback (most recent call last):
      File "/tmp/pip-build-qo8wd315/llvmlite/ffi/build.py", line 220, in <module>
        main()
      File "/tmp/pip-build-qo8wd315/llvmlite/ffi/build.py", line 210, in main
        main_posix('linux', '.so')
      File "/tmp/pip-build-qo8wd315/llvmlite/ffi/build.py", line 134, in main_posix
        raise RuntimeError(msg) from None
    RuntimeError: Could not find a `llvm-config` binary. There are a number of reasons this could occur, please see: https://llvmlite.readthedocs.io/en/latest/admin-guide/install.html#using-pip for help.
    error: command '/mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3' failed with exit status 1

    ----------------------------------------
Command "/mnt/c/Users/Administrator/Documents/parallell2/psc/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qo8wd315/llvmlite/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-ot_v27hb-record/install-record.txt --single-version-externally-managed --compile --install-headers /mnt/c/Users/Administrator/Documents/parallell2/psc/include/site/python3.6/llvmlite" failed with error code 1 in /tmp/pip-build-qo8wd315/llvmlite/`

**Comment by wence-** [comment, marked as answer]
This looks like a problem installing numba. If I follow the link https://llvmlite.readthedocs.io/en/latest/admin-guide/install.html#using-pip, it suggests that you might get around things by `pip install -U pip` and then retrying. Alternately, you can remove the dependency on numba in `setup.py` (it's only there because you might want to use it).

**Reply by joyalrj22** [reply]
This has solved the issue, thank you
